staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -527,6 +527,8 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 
 	s.listedPathProvider = routes.ListedPathProviders{s.listedPathProvider, delegationTarget}
 
+	s.openAPIGetter = apiopenapi.NewSchemaCacher(s.UnprotectedHandler(), routes.OpenAPIv2Path)
```

Comment:
In the configuration where multiple generic apiservers are chained, I'd like to be confident that this handler is *only* going to return openapi types for this apiserver and not for any of the delegate (or parent) apiservers. Can we document that if we are confident, and figure out how to be confident if we aren't?

Reply:
This is the last thing left to do before this PR is done. I still need to figure out how to justify in the code (as a comment) that this isn't going to return openapi types for any delegate or parent apiservers. However I am confident that it doesn't have the undesired behavior because I ran the apiserver locally with a modified get handler which outputs the entire spec stored in the specCacher, and CustomResourceDefinition does not appear in it, but when I request the spec from the /openapi/v2 endpoint, the spec does include CRD.

Reply:
Now I am thinking s.Handler.GoRestfulContainer will be a safer handler

Reply:
Okay I documented this now